### PR TITLE
Add Multi-Object Delete API support (Fixes #413)

### LINKED ISSUE #413: API: Implement RemoveObjects

### PR DESCRIPTION
**Pull request by donatello**
* Adds new `remove_objects` method that performs multi-object delete.
* Adds unit tests.
* Adds functional tests.
* Adds documentation and example.

### CODE REVIEW

**Comment by harshavardhana** [comment]
Can you add a functional test ?

**Comment by donatello** [comment]
@harshavardhana Addressed your review comments - though I am not sure where or how they got lost now.

This is ready for review and merge.

File: minio/definitions.py [hunk 1]
```diff
@@ -176,3 +176,25 @@ def __str__(self):
                                     self.etag,
                                     self.last_modified,
                                     self.size)
+
+
+class MultiDeleteError(object):
```
Comment (harshavardhana) [inline minio/definitions.py]: BTW you can keep this in `error.py`

Comment (donatello) [inline minio/definitions.py]: Done.

File: minio/definitions.py [hunk 2]
```diff
@@ -176,3 +176,25 @@ def __str__(self):
...
+    """
+    Represents an error raised when trying to delete an object in a
+    Multi-Object Delete API call :class:`MultiDeleteError <MultiDeleteError>`
+
+    :object_name: Object name that had the delete error.
```
Comment (donatello) [inline minio/definitions.py]: Fixed.

**Pull request merged**